cpp/bench/ann/src/raft/fix_latency_workload.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cassert>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <raft/core/logger.hpp>
+#include <raft/core/resource/cuda_stream.hpp>
+#include <raft/distance/distance_types.hpp>
+#include <stdexcept>
+#include <string>
+
+#include <raft/util/cudart_utils.hpp>
+
+#include <chrono>
+#include <thread>
+
+namespace raft::bench::ann {
+
+// This kernel sleeps for 10ms
+__global__ inline void kernel_sleep(int sleep_ms)
+{
+  for (int i = 0; i < sleep_ms; i++)
+    __nanosleep(1000000U);  // lms
+}
+
+inline void workload(bool use_gpu, bool sync_stream, int sleep_ms, cudaStream_t stream)
+{
+  if (use_gpu) {
+    kernel_sleep<<<1, 1, 0, stream>>>(sleep_ms);
+    if (sync_stream) { cudaStreamSynchronize(stream); }
+  } else {
+    std::this_thread::sleep_for(std::chrono::milliseconds(sleep_ms));
+  }
+}
+
+class FixLatencyWorkload : public ANN<float> {
```

Comment:
This is just a tool to test the timers. Shall we remove before merging the PR?

Reply:
Yeah, I don't think we need to keep this.

Reply:
removed